x/ccv/provider/keeper/key_assignment.go

```diff
@@ -566,6 +566,8 @@ func (k Keeper) MustApplyKeyAssignmentToValUpdates(
 
 // GetProviderAddrFromConsumerAddr returns the consensus address of a validator with
 // consAddr set as the consensus address on a consumer chain
+//
+// TODO: communicate better that this method may return a consumer addr
```

Comment:
What would precipitate this function returning a consumer address?

Reply:
```golang
	// check if this address is known only to the consumer chain
	if providerConsAddr, found := k.GetValidatorByConsumerAddr(ctx, chainID, consAddr); found {
		return providerConsAddr
	}
	return consAddr
```

As it stands now, it is a bit messy.

Consumer address is equal to provider address in case the operator does not use key assignment (ie, they use the same validator key for provider and consumer)

Reply:
@MSalopek has the correct understanding. That code behavior was not clear to me from reading code/comments